libraries/standard/json/src/json.c

```diff
@@ -41,11 +43,13 @@ static void skipSpace( const char * buf,
                        size_t * start,
                        size_t max )
 {
-    size_t i = *start;
+    size_t i;
+
+    assert( ( buf != NULL ) && ( start != NULL ) && ( max > 0U ) );
```

Comment:
`assert(__CPROVER_r_ok(buf,*start+max))` - or something similar

Reply:
I'd like to keep the proofs separate from the actual library, and not entangle things.

Reply:
In some libraries we've worked on, they've added something like

```C
#ifdef CBMC
#   define PTR_IS_VALID(p,l) (l == 0 || __CPROVER_r_ok(p,l)
#else
#   define PTR_IS_VALID(p,l) (l == 0 || ptr != NULL)
#endif
```

Would something like that make sense here?

Reply:
If I were a customer reading this code, I think I would be more reassured by the familiar `assert( ptr != NULL)`.